network/crl.go

```diff
@@ -70,6 +71,8 @@ const (
 )
 
 const (
+	// Max int value, because cache implementation uses `int` for setting capacity
+	crlCacheSizeMax = 2147483647
```

Comment:
Uh...

1. If we are going to use `int` why read it as `uint` in the first place? And then bother with conversions here and there, sometimes even not checking ranges.
2. The spec says that `int` might be either 32 or 64-bit. Reliable way to get its maximum value is something like `int(^uint(0) >> 1)`.

If you'd like to avoid negative values, I believe that `size >= 0` is much simpler to write than `size <= crlCacheSizeMax` which you still have to do.

Reply:
Making the flag itself `Uint` + converting to `int` is more obvious, it will detect error earlier and negative cache size does not make sense anyway. That's why I thought it was better than `Int` + validation it's `> 0`. Also, even max value of 32-bit int is overkill for cache size.

Reply:
https://golang.org/pkg/math/#pkg-constants - math.MaxInt8|16|32|64

Reply:
Well, `math` constants exist but you still don't know whether `int` is `int32` or `int64`.

FWIW, we can say
```go
const crlCacheSizeMax = 1_000_000
```
or any other 'reasonable' number, for a certain definition thereof, as long as `crlCacheSizeMax <= math.MaxInt32`.

What I don't really want is to trade a single `if` in option/config parsing code for a whole bunch of range checks and random type casts after that. For example, slice length in Go is signed `int`. While arguably it should have been unsigned for the sake of domain correctness, the world didn't burn down because of that.

Reply:
Ah, took `1_000_000` as max value. Nobody would ever need such a big cache anyway (where every cache item corresponds to unique URL, so old CRLs won't take space)